src/main/java/org/opensearch/neuralsearch/processor/CompoundTopDocs.java

```diff
@@ -99,14 +104,25 @@ public CompoundTopDocs(final QuerySearchResult querySearchResult) {
         // skipping first two elements, it's a start-stop element and delimiter for first series
         List<TopDocs> topDocsList = new ArrayList<>();
         List<ScoreDoc> scoreDocList = new ArrayList<>();
+        List<Object> collapseValueList = new ArrayList<>();
         for (int index = 2; index < scoreDocs.length; index++) {
             // getting first element of score's series
             ScoreDoc scoreDoc = scoreDocs[index];
             if (isHybridQueryDelimiterElement(scoreDoc) || isHybridQueryStartStopElement(scoreDoc)) {
                 ScoreDoc[] subQueryScores = scoreDocList.toArray(new ScoreDoc[0]);
                 TotalHits totalHits = new TotalHits(subQueryScores.length, TotalHits.Relation.EQUAL_TO);
                 TopDocs subQueryTopDocs;
-                if (isSortEnabled) {
+                if (isCollapseEnabled) {
```

Comment:
```suggestion
                if (isCollapseEnabled && topDocs instanceof CollapseTopFieldDocs collapseDocs) {
```
then you can remove L116

Reply:
This check already happens when isCollapseEnabled is set:
```
if (topDocs instanceof CollapseTopFieldDocs) {
            isCollapseEnabled = true;
        } else if (topDocs instanceof TopFieldDocs) {
            isSortEnabled = true;
        }
```

Therefore, if we add it to the if statement then the second half of the check would always be true.